### PR TITLE
support auth

### PR DESCRIPTION
**Pull request by weiliu1031**
Signed-off-by: Wei Liu <wei.liu@zilliz.com>
Issue: https://github.com/milvus-io/milvus/issues/18228

### CODE REVIEW

File: internal/util/paramtable/service_param.go [hunk 1]
```diff
@@ -369,6 +382,45 @@ func (p *PulsarConfig) initMaxMessageSize() {
...
+	}
+
+	if len(jsonMap) == 0 {
+		p.AuthParams = ""
+	} else {
+		jsonData, _ := json.Marshal(&jsonMap)
+		p.AuthParams = string(jsonData)
+	}
+	pulsarOnce.Do(func() {
+		cmdutils.PulsarCtlConfig.AuthParams = p.AuthParams
```
Comment (jiaoew1991) [inline internal/util/paramtable/service_param.go:412]: I think it is a bad design to modify variables in `cmdutils` at here

File: internal/mq/msgstream/mqwrapper/pulsar/pulsar_client.go [hunk 2]
```diff
@@ -100,6 +115,18 @@ func (pc *pulsarClient) Subscribe(options mqwrapper.ConsumerOptions) (mqwrapper.
 	return pConsumer, nil
 }
 
+func GetFullTopicName(tenant string, namespace string, topic string) (string, error) {
```
[the remark this answers is not in the document]
Comment (weiliu1031) [reply internal/mq/msgstream/mqwrapper/pulsar/pulsar_client.go:118]: no, for which doesn't specified tenant/namespace, it actually used the default tenant `public` and namespace 'default', so nothing changed.

File: internal/mq/msgstream/mq_factory.go [hunk 3]
```diff
@@ -65,13 +84,34 @@ func (f *PmsFactory) NewMsgStream(ctx context.Context) (MsgStream, error) {
...
 	return NewMqTtMsgStream(ctx, f.ReceiveBufSize, f.PulsarBufSize, pulsarClient, f.dispatcherFactory.NewUnmarshalDispatcher())
 }
 
+func (f *PmsFactory) getAuthentication() (pulsar.Authentication, error) {
+	auth, err := pulsar.NewAuthentication(f.PulsarAuthPlugin, f.PulsarAuthParams)
+
+	if err != nil {
+		log.Error("build authencation from config failed, please check it!",
+			zap.String("authPlugin", f.PulsarAuthPlugin),
+			zap.String("authParams", f.PulsarAuthParams))
```
Comment (congqixia) [inline internal/mq/msgstream/mq_factory.go:109]: Maybe we need add `zap.Error(err)` to show the exact error message from pulsar here.

File: internal/util/paramtable/service_param.go [hunk 4]
```diff
@@ -369,6 +382,45 @@ func (p *PulsarConfig) initMaxMessageSize() {
...
+	})
+}
+
+func (p *PulsarConfig) initTenant() {
+	p.Tenant = p.Base.LoadWithDefault("pulsar.tenant", "public")
+}
+
+func (p *PulsarConfig) initNamespace() {
+	p.Namespace = p.Base.LoadWithDefault("pulsar.amespace", "default")
```
Comment (congqixia) [inline internal/util/paramtable/service_param.go:421]: typo: `pulsar.namespace`

File: internal/util/paramtable/service_param.go [hunk 5]
```diff
@@ -369,6 +382,45 @@ func (p *PulsarConfig) initMaxMessageSize() {
...
+
+	// parse auth params to json, auth params example: token: fake_token
+	jsonMap := make(map[string]string)
+	params := strings.Split(paramString, ",")
+	for _, param := range params {
+		kv := strings.Split(param, ":")
+		if len(kv) == 2 {
+			jsonMap[kv[0]] = kv[1]
+		}
+	}
```
Comment (congqixia) [inline internal/util/paramtable/service_param.go:403]: do we have an example for authParams? this process looks too complicated here.

**Comment by weiliu1031** [comment]
rerun ut

File: internal/mq/msgstream/mq_factory.go [hunk 6]
```diff
@@ -81,9 +119,16 @@ func (f *PmsFactory) NewQueryMsgStream(ctx context.Context) (MsgStream, error) {
...
 	return func(channels []string, subname string) error {
 		// try to delete the old subscription
-		admin := cmdutils.NewPulsarClient()
+		admin, err := puslarmqwrapper.NewAdminClient(f.PulsarWebAddress, f.PulsarAuthPlugin, f.PulsarAuthParams)
+		if err != nil {
+			return err
+		}
 		for _, channel := range channels {
-			topic, err := utils.GetTopicName(channel)
+			fullTopicName, err := puslarmqwrapper.GetFullTopicName(f.PulsarTenant, f.PulsarNameSpace, channel)
```
Comment (jiaoew1991) [inline internal/mq/msgstream/mq_factory.go:127]: pulsar

File: internal/util/paramtable/service_param.go [hunk 7]
```diff
@@ -369,6 +375,41 @@ func (p *PulsarConfig) initMaxMessageSize() {
...
+	if len(jsonMap) == 0 {
+		p.AuthParams = ""
+	} else {
+		jsonData, _ := json.Marshal(&jsonMap)
+		p.AuthParams = string(jsonData)
+	}
+}
+
+func (p *PulsarConfig) initTenant() {
+	p.Tenant = p.Base.LoadWithDefault("pulsar.tenant", "public")
```
Comment (jiaoew1991) [inline internal/util/paramtable/service_param.go:406]: default value should add into milvus.yaml

File: internal/mq/msgstream/mq_factory.go (hunk 6 above)
Comment (weiliu1031) [reply internal/mq/msgstream/mq_factory.go:127]: next pr fix this

**Review by congqixia — APPROVED** [review]
default value and misusing value shall be fixed in next pr
/lgtm

**Pull request merged**